assets/src/components/detours/detourMap.tsx

```diff
@@ -0,0 +1,204 @@
+import React, { useState } from "react"
+import { Shape } from "../../schedule"
+import { LatLngExpression } from "leaflet"
+import { Polyline, useMap, useMapEvent } from "react-leaflet"
+import Leaflet, { Map as LeafletMap } from "leaflet"
+import Map from "../map"
+import { CustomControl } from "../map/controls/customControl"
+import { Button } from "react-bootstrap"
+import { ReactMarker } from "../map/utilities/reactMarker"
+
+export const DetourMap = ({ shape }: { shape: Shape }) => {
+  const [startPoint, setStartPoint] = useState<LatLngExpression | null>(null)
+  const [endPoint, setEndPoint] = useState<LatLngExpression | null>(null)
+  const [detourPositions, setDetourPositions] = useState<LatLngExpression[]>([])
+
+  const onAddDetourPosition = (p: LatLngExpression) =>
+    setDetourPositions((positions) => [...positions, p])
+
+  return (
+    <Map vehicles={[]}>
+      <CustomControl position="topleft" className="leaflet-bar">
+        <Button
+          variant="primary"
+          disabled={
+            startPoint === null ||
+            endPoint !== null ||
+            detourPositions.length === 1
+          }
+          onClick={() =>
+            setDetourPositions((positions) =>
+              positions.slice(0, positions.length - 1)
+            )
+          }
+        >
+          Clear Last Waypoint
+        </Button>
+      </CustomControl>
+      <RouteShapeWithDetour
+        shape={shape}
+        startPoint={startPoint}
+        onSetStartPoint={setStartPoint}
+        endPoint={endPoint}
+        onSetEndPoint={setEndPoint}
+        detourPositions={detourPositions}
+        onAddDetourPosition={onAddDetourPosition}
+      />
+    </Map>
+  )
+}
+
+const RouteShapeWithDetour = ({
+  shape,
```

Comment:
If `shape` here means "the original un-diverted route shape", can we call it `original` or `originalShape` or something like that?

Reply:
Yeah, I like `originalShape`, I think I'll go with that.

Reply:
55ba0b74